File: src/main/java/net/matees/itemrush/settings/RandomItemCount.java
package net.matees.itemrush.settings;

import net.matees.settings.BooleanSetting;

import java.util.Random;

import org.bukkit.Material;

public class RandomItemCount extends BooleanSetting {

    public Boolean randomItemCount = true;

    private static final RandomItemCount INSTANCE = new RandomItemCount();

    private RandomItemCount() {
    }

    public static RandomItemCount getInstance() {
        return INSTANCE;
    }

    @Override
    public String getName() {
        return "Random Item Count";
    }

    @Override
    public String getDescription() {
        return "Toggle random amount of items dropped";
    }

    @Override
    public Boolean getSetting() {
        return this.randomItemCount;
    }

    @Override
    public void setSetting(Boolean setting) {
        this.randomItemCount = setting;
    }

    @Override
    public Material getMenuItemMaterial() {
        return Material.REDSTONE_BLOCK;
    }

    @Override
    public int getMenuItemSlot() {
        return 11;
    }
}

File: src/main/java/net/matees/arcade/itemrush/settings/RandomItemCount.kt
package net.matees.arcade.itemrush.settings

import net.matees.settings.BooleanSetting
import org.bukkit.Material

class RandomItemCount private constructor() : BooleanSetting() {
    var randomItemCount: Boolean = true

    override val name: String
        get() = "Random Item Count"

    override val description: String
        get() = "Toggle random amount of items dropped"

    override var setting: Boolean?
        get() = this.randomItemCount
        set(setting) {
            if (setting != null) {
                this.randomItemCount = setting
            }
        }

    override val menuItemMaterial: Material
        get() = Material.REDSTONE_BLOCK

    override val menuItemSlot: Int
        get() = 11

    companion object {
        val instance: RandomItemCount = RandomItemCount()
    }
}
